### PR TITLE
Optimize `toggleSeenStatus` realm usage

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #352

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt [hunk 1]
```diff
@@ -118,32 +100,53 @@ object ThreadController {
...
+        val mailboxUuid = MailboxController.getCurrentMailboxUuid() ?: return
+
+        RealmDatabase.mailboxContent().writeBlocking {
+            val latestThread = findLatest(thread) ?: return@writeBlocking
+            val uids = getThreadUnseenMessagesUids(latestThread)
+            with(ApiRepository.markMessagesAsSeen(mailboxUuid, uids)) {
+                if (isSuccess()) markThreadAsSeen(latestThread)
+            }
         }
     }
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt:131]: I think we could factor better, there is some duplication

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt [hunk 2]
```diff
@@ -118,32 +100,53 @@ object ThreadController {
...
-        val uids = getThreadUnseenMessagesUids(latestThread)
-        with(ApiRepository.markMessagesAsSeen(latestThread.mailboxUuid, uids)) {
-            if (isSuccess()) realm.markThreadAsSeen(latestThread, folderId)
+    private fun MutableRealm.markThreadAsSeen(thread: Thread) {
+        thread.messages.forEach {
+            if (!it.seen) {
+                incrementFolderUnreadCount(it.folderId, -1)
+                it.seen = true
+            }
         }
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt:147]: You could call `incrementFolderUnreadCount` after the loop

**Review by JorisBodin — APPROVED** [review]

**Review by JorisBodin — CHANGES_REQUESTED** [review]
.

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt (hunk 2 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt:147]: The `folderId` is different for each `Message`, so we need to do it in the loop.

**Review by sirambd — APPROVED** [review]

**Review by JorisBodin — APPROVED** [review]

**Pull request merged**